test/e2e/ai/conversation_summarization_test.go

```diff
@@ -0,0 +1,237 @@
+//go:build e2e
+
+package ai
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/anthropics/anthropic-sdk-go"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cchalm/blundering-savant/internal/ai"
+	"github.com/cchalm/blundering-savant/test/e2e/testutil"
+)
+
+// TestConversationSummarization tests that AI can understand its own summaries
+func TestConversationSummarization(t *testing.T) {
+	harness := testutil.NewTestHarness(t)
+
+	testName := "conversation_summarization_understanding"
+	harness.RunIterations(testName, func(iteration int) error {
+		return harness.WithTimeout(func(ctx context.Context) error {
+			conversation, err := harness.CreateConversationWithSystemPrompt("Blundering Savant", "blunderingsavant")
+			require.NoError(t, err)
+
+			// Step 1: Build a complex conversation history
+			initialMessage := `Repository: test/web-app
+
+Main Language: JavaScript
+
+## Issue
+
+Issue #456: Implement user authentication system
+
+### Description
+
+> We need to implement a complete user authentication system with registration, login, logout, and password reset functionality. The system should use JWT tokens and integrate with our existing user database.
+
+## Your Task
+
+Please analyze the requirements and create a plan for implementing the authentication system. Consider security best practices and scalability.`
+
+			// Send initial message to establish context
+			response1, err := conversation.SendMessage(ctx, anthropic.NewTextBlock(initialMessage))
+			if err != nil {
+				return fmt.Errorf("failed to send initial message: %w", err)
+			}
+
+			// Add several more exchanges to build conversation history
+			followup1 := `Great analysis! Now please start by implementing the user registration endpoint. Make sure to include proper input validation and password hashing.`
```

Comment:
Do not use the AI to build this conversation history - it's expensive and makes things less deterministic than necessary. Instead, manually construct a hard-coded conversation history. Consider using conversation resumption to do this, if the conversation type doesn't export the necessary functionality otherwise